src/pages/productPage.ts

```diff
@@ -73,8 +74,15 @@ export default class ProductPage extends BasePage {
     });
   }
 
-  // Verify methods
+  async clickOnAddToWishListButton() {
+    await test.step('Click on "Add to wishlist" button', async () => {
+      await this.waitUntilLoad(this.PAGE_STATE.DOM_CONTENT_LOADED);
+      await this.page.waitForTimeout(1000);
```

Comment:
Don't use explicit waits in the code. It's a bad practice. Instread  try waitFor or expect toBeVisible
Docs links: [https://playwright.dev/docs/api/class-locator#locator-wait-for](url) - waitFor
[https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-be-visible](url) - toBe Visible